### PR TITLE
DAT-590 cherry picker

### PR DESCRIPTION
**Pull request by zenlan**
Some cherries extracted from a recent code review in preparation for DAT-590 - the implementation of stats calculations for unpowered items, new categories and new lca data.

[This document](https://docs.google.com/document/d/11ANtNz8bnmxJ2SLnfApVRswrRQ8kdcHSi1sxCxQ5vNU/edit?usp=sharing) has an overview of changes along with routes and pages for manual testing.

### CODE REVIEW

File: app/Group.php [hunk 1]
```diff
@@ -249,9 +248,7 @@ public function canDelete()
             ->get();
 
         foreach ($allEvents as $event) {
-            $footprintRatioCalculator = new FootprintRatioCalculator();
-            $emissionRatio = $footprintRatioCalculator->calculateRatio();
-            $stats = $event->getEventStats($emissionRatio);
+            $stats = $event->getEventStats();
```
Comment (ngm) [inline app/Group.php:251]: Note to self: if this is null, it's now calculated in getEventStats.

File: resources/views/group/view.blade.php [hunk 2]
```diff
@@ -40,73 +40,51 @@
               $group_image->image->path;
           }
 
-          $can_edit_group = App\Helpers\Fixometer::hasRole( $user, 'Administrator') || $isCoordinatorForGroup || $is_host_of_group;
-          $can_see_delete = App\Helpers\Fixometer::hasRole( $user, 'Administrator');
+          $can_edit_group = App\Helpers\Fixometer::hasRole($user, 'Administrator') || $isCoordinatorForGroup || $is_host_of_group;
+          $can_see_delete = App\Helpers\Fixometer::hasRole($user, 'Administrator');
           $can_perform_delete = $can_see_delete && $group->canDelete();
 
-          $expanded_events = [];
```
Comment (ngm) [inline resources/views/group/view.blade.php:47]: Is expanded_events no longer needed?

**Review by ngm — COMMENTED** [review]
Looks good!

In summary to me, it looks like benign changes that simplify a lot of the code around the stats.

Primarily:

- changes to FootprintRatioCalculator, to make calculateRatio static, and removing the need to call it in a lot of places, favouring moving it inside getGroupStats and getEventStats
- removal of lots of old dead cruft
- addition of tests

I'll now run the phpunit tests locally, and manually test the various routes.

File: resources/views/group/view.blade.php (hunk 2 above)
Comment (zenlan) [reply resources/views/group/view.blade.php:47]: Assignment has been moved to the controller and is still used by the Vue element:
`:events="{{ json_encode($expanded_events, JSON_INVALID_UTF8_IGNORE) }}"`

**Review by ngm — APPROVED** [review]
Difference in some stats as per https://therestartproject.atlassian.net/browse/DAT-590?focusedCommentId=15298 have been resolved.

Develop merged in and conflicts resolved.

LGTM! :+1:

**Pull request merged**